docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,357 @@
+---
+layout: docs
+title: Inject Scripts into Tested Pages
+permalink: /documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.html
+---
+# Inject Scripts into Tested Pages
+
+TestCafe allows you to inject custom client scripts into pages visited during the tests. You can add scripts that mock browser API or provide helper functions.
+
+Use [test run options](#add-client-scripts-to-all-tests) to add client scripts to all tests, or [test API](#add-client-scripts-to-specific-tests) to add them to specific fixtures or tests.
+
+## Add Client Scripts to All Tests
+
+Use either of the following to inject scripts into pages visited during all the tests:
+
+* the [--cs (--client-scripts)](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2) command line option
+
+    ```sh
+    testcafe chrome test.js --client-scripts=mockDate.js,assets/react-helpers.js
+    ```
+
+* the [runner.clientScripts](../programming-interface/runner.md#clientscripts) API method
+
+    ```js
+    runner.clientScripts('mockDate.js', 'scripts/react-helpers.js');
+    ```
+
+* the [clientScripts](../configuration-file.md#clientscripts) configuration file property
+
+    ```json
+    {
+        "clientScripts": ["mockDate.js", "scripts/react-helpers.js"]
+    }
+    ```
+
+## Add Client Scripts to Specific Tests
+
+Use the [fixture.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) and [test.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) methods to inject scripts into pages visited during a particular test or fixture.
```

Comment:
One more note:
The scripts specified in `fixture.clientScripts` are executed first, the scripts from `test.clientScripts` - next.
This is actual for `fixture.requestHooks` and `test.requestHooks` too.